src/netstring_protocol.cpp

```diff
@@ -172,3 +172,14 @@ size_t netstring_client_t::stream_responses(const void* message, size_t size, bo
 }
 
 } // namespace prime_server
+
+prime_server::shortcircuiter_t<prime_server::netstring_entity_t> make_shortcircuiter(const std::string& health_check_str){
+  return [health_check_str](const prime_server::netstring_entity_t& request) {
+    if(request.body == health_check_str){
+      netstring_entity_t response = netstring_entity_t::from_string("healthy", 7);
+      std::string response_str = response.to_string();
+      return std::unique_ptr<zmq::message_t>(new zmq::message_t(response_str.size(), response_str.c_str()));
+    }
+    return std::unique_ptr<zmq::message_t>();
```

Comment:
returning `nullptr` feels more idiomatic, though technically it will be converted to a smart pointer before return, it just reads better imho

Reply:
Done